test/test_specifications.py

```diff
@@ -136,3 +137,13 @@ def raise_exception(_):
     specification._rules[0].verify.assert_called_once()
     specification._rules[1].verify.assert_called_once()
     specification._rules[2].verify.assert_not_called()
+
+
+def test_specification_ensures_rule_order_respected(specification):
+    specification._rules = [IrisReductionRule(True), MolSpecStitchRule(True), IrisCalibrationRule({})]
+
+    specification._order_rules()
+
+    assert specification._rules[0] == IrisReductionRule(True)
+    assert specification._rules[1] == IrisCalibrationRule({})
+    assert specification._rules[-1] == MolSpecStitchRule(True)
```

Comment:
Tests for the inter stitch rule starting in different positions might indicate issues (assuming MolSpecStitchRule is an inter rule) https://github.com/fiaisis/run-detection/blob/72d7334e14779eef839d61ec37d9a30c88456ed9/rundetection/rules/inter_rules.py#L11-L16 , i.e. randomise the order of the rules and see if they are still respected.

Reply:
Inter Stitch is a left over rule from initial prototyping that will have value when we revisit the INTER instrument. INTER is not dependent on MolSpecStitch. The IRIS rules ending up in different positions because we moved MolSpecStitch makes sense here, as it needs to be at the end, so everything after it moves up by 1.